Src/Couchbase.Linq/QueryGeneration/MethodCallTranslators/DictionaryContainsKeyMethodCallTranslator .cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Reflection;
+namespace Couchbase.Linq.QueryGeneration.MethodCallTranslators
+{
+    internal class DictionaryContainsKeyMethodCallTranslator  : IMethodCallTranslator
+    {
+        private static readonly MethodInfo[] SupportedMethodsStatic = {
+            typeof (IDictionary).GetMethod("Contains"),
+            typeof (IDictionary<,>).GetMethod("ContainsKey")
+        };
+
+        public IEnumerable<MethodInfo> SupportMethods => SupportedMethodsStatic;
```

Comment:
The static and instance properties could be combined like this:

```
public static readonly IEnumerable<MethodInfo> SupportedMethods = new [] {
    typeof (IDictionary).GetMethod("Contains"),
    typeof (IDictionary<,>).GetMethod("ContainsKey")
};
```

Reply:
Right now it's an instance method so it can be exposed on the interface, which allows us to get it without so much reflection: https://github.com/couchbaselabs/Linq2Couchbase/blob/master/Src/Couchbase.Linq/QueryGeneration/DefaultMethodCallTranslatorProvider.cs#L36

We could change it, but it would require a large refactor and should probably be a separate issue.

Reply:
Sure, ok - makes sense.